lib/db/schema/patch-051-052.sql

```diff
@@ -0,0 +1,313 @@
+  ) AND isPrimary = true;
+
+  DELETE FROM signinCodes
+  WHERE hash = hashArg
+  AND createdAt > newerThan;
+
+  COMMIT;
+END;
+
+CREATE PROCEDURE `createEmail_2` (
+    IN `normalizedEmail` VARCHAR(255),
+    IN `email` VARCHAR(255),
+    IN `uid` BINARY(16) ,
+    IN `emailCode` BINARY(16),
+    IN `isVerified` TINYINT(1),
+    IN `verifiedAt` BIGINT UNSIGNED,
+    IN `createdAt` BIGINT UNSIGNED
+)
+BEGIN
+    DECLARE EXIT HANDLER FOR SQLEXCEPTION
+    BEGIN
+        ROLLBACK;
+        RESIGNAL;
+    END;
+
+    START TRANSACTION;
+
+    INSERT INTO emails(
+        normalizedEmail,
+        email,
+        uid,
+        emailCode,
+        isVerified,
+        isPrimary,
+        verifiedAt,
+        createdAt
+    )
+    VALUES(
+        LOWER(normalizedEmail),
+        email,
+        uid,
+        emailCode,
+        isVerified,
+        false,
+        verifiedAt,
+        createdAt
+    );
+
+    COMMIT;
+END;
+
+CREATE PROCEDURE `deleteEmail_2` (
+    IN `inUid` BINARY(16),
+    IN `inNormalizedEmail` VARCHAR(255)
+)
+BEGIN
+    SET @primaryEmailCount = 0;
+
+    -- Don't delete primary email addresses
+    SELECT COUNT(*) INTO @primaryEmailCount FROM emails WHERE normalizedEmail = inNormalizedEmail AND uid = inUid AND isPrimary = true;
+    IF @primaryEmailCount = 1 THEN
+        SIGNAL SQLSTATE '45000' SET MYSQL_ERRNO = 2100, MESSAGE_TEXT = 'Can not delete a primary email address.';
+    END IF;
+
+    DELETE FROM emails WHERE normalizedEmail = inNormalizedEmail AND uid = inUid AND isPrimary = false;
+END;
+
+CREATE PROCEDURE `keyFetchTokenWithVerificationStatus_2` (
+  IN `tokenIdArg` BINARY(32)
+)
+BEGIN
+  SELECT
+    t.authKey,
+    t.uid,
+    t.keyBundle,
+    t.createdAt,
+    e.isVerified AS emailVerified,
+    a.verifierSetAt,
+    ut.tokenVerificationId
+  FROM keyFetchTokens AS t
+  LEFT JOIN accounts AS a
+    ON t.uid = a.uid
+  LEFT JOIN emails AS e
+    ON t.uid = e.uid
+  LEFT JOIN unverifiedTokens AS ut
+    ON t.tokenId = ut.tokenId
+  WHERE t.tokenId = tokenIdArg
+  AND e.isPrimary = true;
+END;
+
+CREATE PROCEDURE `passwordChangeToken_3` (
+    IN `inTokenId` BINARY(32)
+)
+BEGIN
+    SELECT
+        t.uid,
+        t.tokenData,
+        t.createdAt,
+        a.verifierSetAt
+    FROM
+        passwordChangeTokens t,
+        accounts a,
+        emails e
+    WHERE
+        t.tokenId = inTokenId
+    AND
+        t.uid = a.uid
+    AND
+        t.uid = e.uid;
+END;
+
+CREATE PROCEDURE `passwordForgotToken_2` (
+    IN `inTokenId` BINARY(32)
+)
+BEGIN
+    SELECT
+        t.uid,
+        t.tokenData,
+        t.createdAt,
+        t.passCode,
+        t.tries,
+        e.email,
+        a.verifierSetAt
+    FROM
+        passwordForgotTokens t,
+        accounts a,
+        emails e
+    WHERE
+        t.tokenId = inTokenId
+    AND
+        t.uid = a.uid
+    AND
+        t.uid = e.uid
+    AND
+        e.isPrimary = true
+    ;
+END;
+
+CREATE PROCEDURE `sessionToken_5` (
+    IN `inTokenId` BINARY(32)
```

Comment:
This one isn't updated in `lib/db/mysql.js`, I just noticed. I can see why your search/replace would have failed, because that file still points at the `_3` version instead of the `_4` version.

Interesting that we never spotted it before, is the basic `sessionToken` stored procedure actually used anywhere, or does everything refer to the `With...` versions?